test/components/alpha/dropdown_test.rb

```diff
@@ -98,4 +99,20 @@ def test_renders_caret
       end
     end
   end
+
+  def test_accepts_custom_values_for_button_aria_label
+    render_inline(Primer::Alpha::Dropdown.new(with_caret: true)) do |component|
+      component.with_button(aria_label_closed: "Open me", aria_label_open: "Close me") { "Button" }
+      component.with_menu do |menu|
+        menu.with_item { "Item" }
+      end
+    end
+
+    assert_selector("details.dropdown") do
+      assert_selector("summary.btn", text: "Button")
+      assert_selector("summary.btn[aria-label='Open me']")
+      assert_selector("summary[data-aria-label-open='Close me']")
+      assert_selector("summary[data-aria-label-closed='Open me']")
+    end
```

Comment:
I think `assert_selector` with an implicit receiver searches the entire page. Calling it on the yielded element instead should have the nesting effect you're after:

```suggestion
    assert_selector("details.dropdown") do |dropdown|
      dropdown.assert_selector("summary.btn", text: "Button")
      dropdown.assert_selector("summary.btn[aria-label='Open me']")
      dropdown.assert_selector("summary[data-aria-label-open='Close me']")
      dropdown.assert_selector("summary[data-aria-label-closed='Open me']")
    end
```